tests/conftest.py

```diff
@@ -323,6 +326,31 @@ def ptfhost(ansible_adhoc, tbinfo, duthost):
         ptf_host = duthost.host.options["inventory_manager"].get_host(duthost.hostname).get_vars()["ptf_host"]
         return PTFHost(ansible_adhoc, ptf_host)
 
+@pytest.fixture(scope="module")
+def k8shosts(ansible_adhoc, request, creds):
+    """
+    Shortcut fixture for getting Kubernetes hosts
+    """
+    master_id = request.config.getoption("--kube_master") 
+    master_id_elements = master_id.split('_')
+    assert len(master_id_elements) == 2, "Invalid master identifier format. Please use format {servernumber}_{mastersetnumber}"
+    master_server_id = master_id_elements[0]
+    master_set_id = master_id_elements[1]
+    k8s_ansible_group = "k8s_vms{}_{}".format(master_set_id, master_server_id)
+    master_vms = {}
+    inv_files = request.config.getoption("ansible_inventory")
+    for inv_file in inv_files:
+        if "k8s" in inv_file:
+            k8s_inv_file = inv_file
+    with open('../ansible/{}'.format(k8s_inv_file), 'r') as kinv:
+        k8sinventory = yaml.safe_load(kinv)
+        for hostname, attributes in k8sinventory[k8s_ansible_group]['hosts'].items():
+            master_vms[hostname[-2:]] = {'host': K8sMasterHost(ansible_adhoc,
```

Comment:
It is better to use full hostname here.

Reply:
I used the last two characters here because I thought it would be cleaner to retrieve the VIP as such: 
```
    haproxy = k8smasters['ha']
    master_vip = haproxy['host'].mgmt_ip
``` 
If I use full hostname, I think I will have to manually create the string `kvm21_1ha`
Would you suggest manually creating the string `kvm21_1ha` based on `run_tests.sh -b` option value, or is there a better way to retrieve the VIP?

Reply:
I think you can add an attribute to the `K8sMasterHost` class to indicate the node type, whether it is a master or ha. Then test scripts can check node type to find out hostname and IP of VIP.

Reply:
In that case, I would need to loop over `k8smasters` to find the `K8sMasterHost` object with `ha` node type to find the VIP instead of indexing with `ha`. Is this right, and suggested over using the last two characters of `hostname` to index? 

For later tests when applying manifests, I was planning to connect to a master host via `k8smasters[m1]`.

Reply:
Indexing with `ha` implicitly assumes that the naming of the k8s nodes in inventory must follow specific rule. People adding new k8s clusters to inventory file may easily neglect this implicit rule. That's why I also suggested to add a class like `K8sCluster` and encapsulate cluster specific information and operations to it, for example, get the VIP, startup/shutdown API server.

Reply:
That makes sense, thank you. I added a `K8sMasterCluster` class and an attribute to help identify `K8sMasterHost` as HAProxy node or not.